xl-v1-namespace.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import "sync"
+
+type nameSpaceParam struct {
+	volume string
+	path   string
+}
+
+type nameSpaceLock struct {
+	rwMutex *sync.RWMutex
+	count   uint
+}
+
+func (nsLock nameSpaceLock) InUse() bool {
+	return nsLock.count != 0
+}
+
+func (nsLock nameSpaceLock) Lock() {
```

Comment:
needs to be a pointer receiver here:
`func (nsLock *nameSpaceLock) Lock() {`

as nsLock.count++ will be lost